compiler/passes/scopeResolve.cpp

```diff
@@ -427,10 +422,22 @@ static void scopeResolve(const AList& alist, ResolveScope* scope) {
         }
       }
 
+    } else if (ForallStmt* forallStmt = toForallStmt(stmt)) {
+      BlockStmt* fBody = forallStmt->loopBody();
+      // or, we could construct ResolveScope specifically for forallStmt
+      ResolveScope* bodyScope = new ResolveScope(fBody, scope);
+      // cf. scopeResolve(FnSymbol*,parent)
+      for_alist(ivdef, forallStmt->inductionVariables()) {
+        Symbol* sym = toDefExpr(ivdef)->sym;
+        if (strcmp(sym->name, "chpl__tuple_blank"))
```

Comment:
Can you explain what's going on with this strcmp? Seems like some sort of weird special case/workaround.
Better would be to find another way, but at a minimum we need a comment talking about the special case it's trying to handle.